.github/workflows/workflow.yaml

```diff
@@ -125,15 +127,19 @@ jobs:
           nix-shell --pure --run 'bazel run //:gazelle-update-repos --config=${{ matrix.ghc-version }}'
           nix-shell --pure --run 'bazel test //... --config=${{ matrix.ghc-version }}'
       - name: Test alternative dependencies
-        run: |
-          cd tests/alternative-deps
-          nix-shell --pure --run 'bazel run //:gazelle --config=${{ matrix.ghc-version }}'
-          nix-shell --pure --run 'bazel run //:gazelle-update-repos --config=${{ matrix.ghc-version }}'
-          nix-shell --pure --run 'bazel test //... --config=${{ matrix.ghc-version }}'
         if: ${{ !matrix.enable-bzlmod }}
+        uses: tweag/run-nix-shell@v0
+        with:
+          working-directory: tests/alternative-deps
+          run: |
+            bazel run //:gazelle --config=${{ matrix.ghc-version }}
+            bazel run //:gazelle-update-repos --config=${{ matrix.ghc-version }}
+            bazel test //... --config=${{ matrix.ghc-version }}
```

Comment:
I think we should add `set -e` here, to make the shell exit immediately when the gazelle commands fail. Maybe we could even make this the default in the run-nix-shell action? Github's `run` step passes `--noprofile --norc -eo pipefail` to `bash` by default.

Reply:
I will add the `set -e`. I agree about having `run-nix-shell` add certain shell flags. I will add this in a subsequent PR.